src/NimBLEAddress.cpp

```diff
@@ -44,25 +48,39 @@ NimBLEAddress::NimBLEAddress(ble_addr_t address) {
  * @param [in] stringAddress The hex representation of the address.
  */
 NimBLEAddress::NimBLEAddress(std::string stringAddress) {
-    if (stringAddress.length() != 17) return;
+    if (stringAddress.length() != 17) {
+        memset(m_address, 0, sizeof m_address); // "00:00:00:00:00:00" represents an invalid address
+        NIMBLE_LOGD(LOG_TAG, "Invalid address '%s'", stringAddress.c_str());
+        return;
+    }
 
     int data[6];
-    sscanf(stringAddress.c_str(), "%x:%x:%x:%x:%x:%x", &data[5], &data[4], &data[3], &data[2], &data[1], &data[0]);
-    m_address[0] = (uint8_t) data[0];
-    m_address[1] = (uint8_t) data[1];
-    m_address[2] = (uint8_t) data[2];
-    m_address[3] = (uint8_t) data[3];
-    m_address[4] = (uint8_t) data[4];
-    m_address[5] = (uint8_t) data[5];
+    if(sscanf(stringAddress.c_str(), "%x:%x:%x:%x:%x:%x", &data[5], &data[4], &data[3], &data[2], &data[1], &data[0]) != 6) {
+        memset(m_address, 0, sizeof m_address); // "00:00:00:00:00:00" represents an invalid address
+        NIMBLE_LOGD(LOG_TAG, "Invalid address '%s'", stringAddress.c_str());
+    }
+    for(size_t index = 0; index < sizeof m_address; index++) {
+        m_address[index] = data[index];
+    }
 } // BLEAddress
 
 
 /**
- * @brief Constructor for compatibility with bluedrioid esp library.
+ * @brief Constructor for compatibility with bluedroid esp library.
  * @param [in] uint8_t[6] or esp_bd_addr_t struct containing the address.
  */
 NimBLEAddress::NimBLEAddress(uint8_t address[6]) {
-    NimBLEUtils::memrcpy(m_address, address, 6);
+    memcpy(m_address, address, sizeof m_address);
```

Comment:
Any reason to not just use `std::reverse_copy` here?

Reply:
There is always just one good reason for not using existing implementations: being unaware of it :D. I'll change it here and in NimBLEUUID too